src/components/test/containers-test.js

```diff
@@ -1,99 +1,70 @@
 import { mount } from 'enzyme';
+import { createRef } from 'preact';
 
 import { Frame, Card, Actions, Scrollbox } from '../containers';
 
-describe('Frame', () => {
-  const createComponent = (props = {}) =>
-    mount(
-      <Frame {...props}>
-        <div>This is content inside of a frame</div>
-      </Frame>
+describe('Container components', () => {
+  const createComponent = (Component, props = {}) => {
+    return mount(
+      <Component {...props}>
+        <div>This is content inside of a container</div>
+      </Component>
     );
-
-  it('renders children inside of a div with appropriate classnames', () => {
-    const wrapper = createComponent();
-
-    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Frame'));
-  });
-
-  it('applies extra classes', () => {
-    const wrapper = createComponent({ classes: 'foo bar' });
-
-    assert.isTrue(wrapper.find('div.Hyp-Frame.foo.bar').exists());
-  });
-});
-
-describe('Card', () => {
-  const createComponent = (props = {}) =>
-    mount(
-      <Card {...props}>
-        <div>This is content inside of a card</div>
-      </Card>
-    );
-
-  it('renders children inside of a div with appropriate classnames', () => {
-    const wrapper = createComponent();
-
-    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Card'));
-  });
-
-  it('applies extra classes', () => {
-    const wrapper = createComponent({ classes: 'foo bar' });
-
-    assert.isTrue(wrapper.find('div.Hyp-Card.foo.bar').exists());
+  };
+
+  const commonTests = (Component, className) => {
+    it('renders children inside of a div with appropriate classnames', () => {
+      const wrapper = createComponent(Component);
+
+      assert.isTrue(wrapper.find('div').first().hasClass(className));
+    });
+
+    it('applies extra classes', () => {
+      const wrapper = createComponent(Component, { classes: 'foo bar' });
+      assert.deepEqual(
+        [
+          ...wrapper
+            .find(`div.${className}.foo.bar`)
+            .getDOMNode()
+            .classList.values(),
+        ],
+        [className, 'foo', 'bar']
+      );
+    });
+
+    it('passes along a `ref` to the input element through `containerRef`', () => {
+      const containerRef = createRef();
+      createComponent(Component, { containerRef });
+
+      assert.isTrue(containerRef.current instanceof Node);
```

Comment:
`assert.instanceOf(object, constructor)` will give a better error if this check fails.

Reply:
Oh, that is a nice assertion to use, thanks.